app/models/service_ansible_playbook.rb

```diff
@@ -44,13 +44,13 @@ def get_job_options(action)
     job_opts = options["#{action.downcase}_job_options".to_sym].deep_dup
     credential_id = job_opts.delete(:credential_id)
     job_opts[:credential] = Authentication.find(credential_id).manager_ref unless credential_id.blank?
-
-    job_opts
+    MiqPassword.decrypt_hash(job_opts)
   end
 
   def save_job_options(action, overrides)
     job_options = parse_dialog_options
-    job_options[:extra_vars] = (job_options[:extra_vars] || {}).merge(overrides[:extra_vars]) if overrides[:extra_vars]
+    job_options[:extra_vars] ||= {}
+    job_options[:extra_vars].merge!(MiqPassword.encrypt_hash(overrides[:extra_vars])) if overrides[:extra_vars]
```

Comment:
you are modifying `job_options[:extra_vars]` in place. you may want `reverse_merge` instead of `merge!`.

To double check this, `puts override_options.inspect` both before and after running this. make sure they are the same